deployment/ccip/changeset/cs_ccip_home.go

```diff
@@ -35,29 +36,127 @@ import (
 	_ deployment.ChangeSet[RevokeCandidateChangesetConfig]       = RevokeCandidateChangeset
 	_ deployment.ChangeSet[UpdateChainConfigConfig]              = UpdateChainConfig
 )
 
+type tokenInfo interface {
+	Address() common.Address
+	Symbol(opts *bind.CallOpts) (string, error)
+	Decimals(opts *bind.CallOpts) (uint8, error)
+}
+
+func findTokenInfo(tokens []tokenInfo, address common.Address) (string, uint8, error) {
+	for _, token := range tokens {
+		if token.Address() == address {
+			tokenSymbol, err := token.Symbol(nil)
+			if err != nil {
+				return "", 0, fmt.Errorf("fetch token symbol for token %s: %w", address, err)
+			}
+			tokenDecimals, err := token.Decimals(nil)
+			if err != nil {
+				return "", 0, fmt.Errorf("fetch token decimals for token %s: %w", address, err)
+			}
+			return tokenSymbol, tokenDecimals, nil
+		}
+	}
+	return "", 0, fmt.Errorf("token %s not found in available tokens", address)
+}
+
 type CCIPOCRParams struct {
 	OCRParameters commontypes.OCRParameters
 	// Note contains pointers to Arb feeds for prices
-	CommitOffChainConfig pluginconfig.CommitOffchainConfig
-	// Note ontains USDC config
-	ExecuteOffChainConfig pluginconfig.ExecuteOffchainConfig
+	CommitOffChainConfig *pluginconfig.CommitOffchainConfig
+	// Note contains USDC config
+	ExecuteOffChainConfig *pluginconfig.ExecuteOffchainConfig
 }
 
-func (c CCIPOCRParams) Validate() error {
+func (c CCIPOCRParams) Validate(selector uint64, feedChainSel uint64, state CCIPOnChainState) error {
 	if err := c.OCRParameters.Validate(); err != nil {
 		return fmt.Errorf("invalid OCR parameters: %w", err)
 	}
-	if err := c.CommitOffChainConfig.Validate(); err != nil {
-		return fmt.Errorf("invalid commit off-chain config: %w", err)
+	if c.CommitOffChainConfig == nil && c.ExecuteOffChainConfig == nil {
+		return errors.New("at least one of CommitOffChainConfig or ExecuteOffChainConfig must be set")
 	}
-	if err := c.ExecuteOffChainConfig.Validate(); err != nil {
-		return fmt.Errorf("invalid execute off-chain config: %w", err)
+	if c.CommitOffChainConfig != nil {
+		if err := c.CommitOffChainConfig.Validate(); err != nil {
+			return fmt.Errorf("invalid commit off-chain config: %w", err)
+		}
+		for tokenAddr, tokenConfig := range c.CommitOffChainConfig.TokenInfo {
+			tokenUnknownAddr, err := ccipocr3.NewUnknownAddressFromHex(string(tokenAddr))
+			if err != nil {
+				return fmt.Errorf("invalid token address %s: %w", tokenAddr, err)
+			}
+			aggregatorUnknownAddr, err := ccipocr3.NewUnknownAddressFromHex(string(tokenConfig.AggregatorAddress))
+			if err != nil {
+				return fmt.Errorf("invalid aggregator address %s: %w", tokenConfig.AggregatorAddress, err)
+			}
+			aggregatorAddr := common.HexToAddress(aggregatorUnknownAddr.String())
+			token := common.HexToAddress(tokenUnknownAddr.String())
+			tokenInfos := make([]tokenInfo, 0)
+			onchainState, ok := state.Chains[selector]
+			if !ok {
+				return fmt.Errorf("chain %d does not exist in state", selector)
+			}
```

Comment:
Can we move this validation to the beginning of this `Validate` method? Would really help in decluttering this code.